src/hasSubObject.ts

```diff
@@ -1,7 +1,9 @@
-import type { Simplify } from "type-fest";
+import type { Merge } from "type-fest";
 import { isDeepEqual } from "./isDeepEqual";
 import { purry } from "./purry";
 
+export type HasSubObjectBrand = { __remeda: "hasSubObject" };
```

Comment:
I haven't deep dived into this PR yet so this isn't a full review.

Shouldn't the branding property use a symbol? It would prevent that property from being part of the object for most use-cases (like JSON serialization, our object functions, etc...).

Reply:
@eranhirsch Yes, I've just been reading about this in [an article](https://egghead.io/blog/using-branded-types-in-typescript) and was going to write about it  :).

Here is an example of how Effect implements brand type - https://github.com/Effect-TS/effect/blob/main/packages/effect/src/Brand.ts#L30-L36

I will change it if you give the green light for the usage of brand type in that PR.

Reply:
If the usage of the branded types doesn't require changes to the tests and they still pass, I'm OK with the change; if this change requires changing the tests to accommodate it, then it's problematic.

Reply:
@eranhirsch By tests, do you mean only type tests, only regular tests or both?

It doesn't require changes for the regular tests. It requires changes for the type tests where we use `toEqualTypeOf` but doesn't where we use `toMatchTypeOf`.

Reply:
Yea, that makes sense. I think those are OK as the `MatchTypeOf` would mean that this isn't a breaking change for anyone.

@cjquines — Using symbols instead of string keys and values makes sense to me. What's your stance here?

Reply:
if symbols were used, then i'm fine with shipping this PR!

Reply:
Then I will update the PR (probably at the beginning of the next week) and will let you know when it's done.

Reply:
@eranhirsch @cjquines Done.